### PR TITLE
Addresses unitary system no load supply air flow rate incorrect sizing report

### LINKED ISSUE #9767: Unitary system doesn't enforce no-load air flow rate
Issue overview
--------------
While working on #6109, I noticed that the `UnitarySystem_MultiSpeedDX.idf` example file (simulated in Seattle) shows that the unitary system doesn't seem to enforce the no-load air flow rate. On the figure below we can see that at the end of each day (the x-axis labels are incorrect, results for two days are shown) both the heating and cooling coils are off, and the outdoor air flow rate is at it's minimum but the mixed air flow rate is about 0.2 m3/s which corresponds to the system's highest speed flow rate according to the EIO file.

image

```
Component Sizing Information, Controller:OutdoorAir, SYS 2 FURNACE DX COOL OA CONTROLLER, Minimum Outdoor Air Flow Rate [m3/s], 4.72000E-002
Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Heating Supply Air Flow Rate [m3/s], 0.20087
Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Cooling Supply Air Flow Rate [m3/s], 0.20087
Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size No Load Supply Air Flow Rate [m3/s], 6.69578E-002
Component Sizing Information, Coil:Cooling:DX:CurveFit:Speed, SYS 2 FURNACE DX COOL COOLING COIL SPEED 1 PERFORMANCE, User-Specified Rated Air Flow Rate [m3/s], 6.69511E-002
Component Sizing Information, Coil:Cooling:DX:CurveFit:Speed, SYS 2 FURNACE DX COOL COOLING COIL SPEED 2 PERFORMANCE, User-Specified Rated Air Flow Rate [m3/s], 0.13392
Component Sizing Information, Coil:Cooling:DX:CurveFit:Speed, SYS 2 FURNACE DX COOL COOLING COIL SPEED 3 PERFORMANCE, User-Specified Rated Air Flow Rate [m3/s], 0.20087
```

### Details
Some additional details for this issue (if relevant):
 - Platform: Windows 10
 - Version of EnergyPlus: 22.2

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by Nigusse**
Pull request overview
---------------------
 - Fixes #9767
 - The `Design Size No Load Supply Air Flow Rate` reported to the EIO file is now corresponds to the user specified or default value for the input field `No Load Supply Air Flow Rate Ratio` in `UnitarySystemPerformance:Multispeed` object. Diffs is expected in EIO files for unitary system example files where `UnitarySystemPerformance:Multispeed` object is specified and the `No Load Supply Air Flow Rate Ratio` field is blank.

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/UnitarySystem.cc [hunk 1]
```diff
@@ -2111,7 +2111,11 @@ namespace UnitarySystems {
                     NoLoadHeatingAirFlowRateRatio =
                         min(this->m_NoLoadAirFlowRateRatio, state.dataUnitarySystems->designSpecMSHP[MSHPIndex].heatingVolFlowRatio[0]);
                 }
-                this->m_NoLoadAirFlowRateRatio = min(NoLoadCoolingAirFlowRateRatio, NoLoadHeatingAirFlowRateRatio);
+                if (state.dataUnitarySystems->designSpecMSHP[MSHPIndex].noLoadAirFlowRateRatio > 0.0) {
+                    this->m_NoLoadAirFlowRateRatio = state.dataUnitarySystems->designSpecMSHP[MSHPIndex].noLoadAirFlowRateRatio;
+                } else {
+                    this->m_NoLoadAirFlowRateRatio = min(NoLoadCoolingAirFlowRateRatio, NoLoadHeatingAirFlowRateRatio);
+                }
```
Comment (Nigusse) [inline src/EnergyPlus/UnitarySystem.cc:2118]: added this logic such that when the `noLoadAirFlowRateRatio` greater than zero (this field is specified by user), then it uses this ratio to calculate the `Design Size No Load Supply Air Flow Rate`.

File: tst/EnergyPlus/unit/UnitarySystem.unit.cc [hunk 2]
```diff
@@ -4029,6 +4029,12 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_MultispeedPerformance)
     EXPECT_NEAR(state->dataUnitarySystems->designSpecMSHP[0].coolingVolFlowRatio[9], 1.0000, 0.00001);
     EXPECT_NEAR(state->dataUnitarySystems->designSpecMSHP[0].heatingVolFlowRatio[9], 1.0000, 0.00001);
 
+    // test system no load air volume flow rate
+    Real64 result1_expected_NoLoadAirVolFlow =
+        state->dataUnitarySystems->designSpecMSHP[0].noLoadAirFlowRateRatio * std::min(thisSys->m_MaxCoolAirVolFlow, thisSys->m_MaxHeatAirVolFlow);
+    EXPECT_EQ(0.05, state->dataUnitarySystems->designSpecMSHP[0].noLoadAirFlowRateRatio);
+    EXPECT_EQ(result1_expected_NoLoadAirVolFlow, thisSys->m_MaxNoCoolHeatAirVolFlow);
+
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/UnitarySystem.unit.cc]: Modified an existing unit test to demonstrate the fix.

File: tst/EnergyPlus/unit/UnitarySystem.unit.cc [hunk 3]
```diff
@@ -4110,6 +4116,12 @@ TEST_F(ZoneUnitarySysTest, UnitarySystemModel_MultispeedPerformance)
               thisSys->m_CoolVolumeFlowRate[10] * state->dataUnitarySystems->designSpecMSHP[0].coolingVolFlowRatio[6]);
     EXPECT_EQ(thisSys->m_HeatVolumeFlowRate[7],
               thisSys->m_HeatVolumeFlowRate[10] * state->dataUnitarySystems->designSpecMSHP[0].heatingVolFlowRatio[6]);
+
+    // test system no load air volume flow rate
+    Real64 result2_expected_NoLoadAirVolFlow =
+        state->dataUnitarySystems->designSpecMSHP[0].noLoadAirFlowRateRatio * std::min(thisSys->m_MaxCoolAirVolFlow, thisSys->m_MaxHeatAirVolFlow);
+    EXPECT_EQ(0.05, state->dataUnitarySystems->designSpecMSHP[0].noLoadAirFlowRateRatio);
+    EXPECT_EQ(result2_expected_NoLoadAirVolFlow, thisSys->m_MaxNoCoolHeatAirVolFlow);
 }
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/UnitarySystem.unit.cc]: Modified the same existing unit test to demonstrate the fix for a different flow rate.

File: tst/EnergyPlus/unit/UnitarySystem.unit.cc [hunk 4]
```diff
@@ -9126,6 +9138,12 @@ TEST_F(EnergyPlusFixture, UnitarySystemModel_MultispeedDXCoilSizing)
     EXPECT_NEAR(1.0 / 10, thisSys->m_MSHeatingSpeedRatio[1] / thisSys->m_MSHeatingSpeedRatio[10], 0.00001);
 
     EXPECT_FALSE(thisSys->m_MyFanFlag); // fan speed variables have been set
+
+    // test system no load air volume flow rate
+    Real64 result1_expected_NoLoadAirVolFlow =
+        state->dataUnitarySystems->designSpecMSHP[0].noLoadAirFlowRateRatio * std::min(thisSys->m_MaxCoolAirVolFlow, thisSys->m_MaxHeatAirVolFlow);
+    EXPECT_EQ(1.0, state->dataUnitarySystems->designSpecMSHP[0].noLoadAirFlowRateRatio);
+    EXPECT_EQ(result1_expected_NoLoadAirVolFlow, thisSys->m_MaxNoCoolHeatAirVolFlow);
 }
```
Comment (Nigusse) [inline tst/EnergyPlus/unit/UnitarySystem.unit.cc]: Modified another existing unit test to demonstrate the fix for a different multi-speed DX cooling coil type and different `No Load Supply Air Flow Rate Ratio` value.

**Review by Nigusse — COMMENTED** [review]
Code change walk through.

**Comment by Nigusse** [comment]
The EIO output snippet shows the `Design Size No Load Supply Air Flow Rate` for develop and and this branch for the defect file. The value in the branch matches the system design flow rate because the `No Load Supply Air Flow Rate Ratio` field is blank, which defaults to 1.0. 
 
Develop:
```
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Supply Air Flow Rate [m3/s], 0.20087
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Heating Supply Air Flow Rate [m3/s], 0.20087
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Cooling Supply Air Flow Rate [m3/s], 0.20087
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size No Load Supply Air Flow Rate [m3/s], 6.69578E-002
```

Branch:
```
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Supply Air Flow Rate [m3/s], 0.20087
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Heating Supply Air Flow Rate [m3/s], 0.20087
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size Cooling Supply Air Flow Rate [m3/s], 0.20087
 Component Sizing Information, AirLoopHVAC:UnitarySystem, SYS 2 FURNACE DX COOL UNITARY SYSTEM, Design Size No Load Supply Air Flow Rate [m3/s], 0.20087
```

**Comment by Nigusse** [comment]
The five test suite idfs have shown diffs. The following four idfs are expected to show diffs in the EIO and html tabular output files due to change in the `Design Size No Load Supply Air Flow Rate`. However, diff is not expected in `SolarShadingTest.idf` file and it must be unrelated diff.
```
EMSLoadBasedMultiSpeedDXCoilOverrideControl.idf (diff is expected)
EMSSetpointBasedMultiSpeedDXCoilOverrideControl.idf (diff is expected)
PythonPluginLoadBasedMultiSpeedDXCoilOverrideControl.idf (diff is expected)
UnitarySystem_MultiSpeedDX.idf (diff is expected)
```

**Review by Myoldmopar — APPROVED** [review]
Simple, minimal, but effective change.  Looks good and glad you were able to just append to existing unit tests, that's great!

**Comment by Myoldmopar** [comment]
Diffs look to be as expected. Good stuff.  Just a final pull from develop and build test locally and this should be ready.

**Comment by Myoldmopar** [comment]
All happy here, merging this one.  Thanks @Nigusse

**Pull request merged**